crates/sound/src/audio_backends.rs

```diff
@@ -9,14 +9,15 @@ mod pw_backend;
 #[cfg(feature = "null-backend")]
 use self::null::NullBackend;
 #[cfg(feature = "pw-backend")]
-use self::pw_backend::{PwBackend, PCMParams};
+use self::pw_backend::PwBackend;
+use crate::vhu_sound::PCMParams;
 use crate::{Error, Result};
 
 pub trait AudioBackend {
     fn write(&self, stream_id: u32) -> Result<()>;
     fn read(&self, stream_id: u32) -> Result<()>;
 
-    fn set_param(&self, _stream_id: u32, _params: PCMParams) -> Result<()> {
+    fn set_param(&mut self, _stream_id: u32, _params: PCMParams) -> Result<()> {
```

Comment:
AudioBackend should in theory be shared by any number of threads, which means it either needs to be behind a Mutex or have interior mutability only

Reply:
Thanks for the comment. If I understand correctly, interior mutability means that internal fields of the , e.g., AudioBacked structure, shall be protected whereas accesing the whole structure won't require protection by itself. I think internal mutability may improve concurrency.